src/llvm/ir.rs

```diff
@@ -123,14 +100,23 @@ impl Metadata {
             | LLVMMetadataKind::LLVMDIStringTypeMetadataKind
             | LLVMMetadataKind::LLVMDIGenericSubrangeMetadataKind
             | LLVMMetadataKind::LLVMDIArgListMetadataKind
-            | LLVMMetadataKind::LLVMDIAssignIDMetadataKind => MetadataKind::Unknown(self),
+            | LLVMMetadataKind::LLVMDIAssignIDMetadataKind => Metadata::Unknown(value),
         }
     }
 }
 
+impl TryFrom<MDNode> for Metadata {
+    type Error = ();
+
+    fn try_from(md_node: MDNode) -> Result<Self, Self::Error> {
+        // FIXME: fail if md_node isn't a Metadata node
```

Comment:
Hmm... I don't think we should really care about that here? Instantiating `MDNode` which is not really a `MDNode` shouldn't be even possible when using `new` methods from our enums safely - they should be responsible for calling `IsAMDNode` and other check methods.